### ISSUE TITLE
Error precision and sample encoding

### ISSUE THREAD

**Issue by pkufool** [lhotse-speech/lhotse#897]
Someone reported to me that the dumped audio is not as expected.

She mixed a speech-cut and noise-cut (both of them are with sampling rate 16K, precision 16bit, sample encoding 16bit), then, she dumped the mixed cut with cutset.save_audios(), but the precison of the saved audio is 25bit and sample encoding is 32bit floating point.

![image](https://user-images.githubusercontent.com/11765074/203058676-6b70e757-481f-403f-ad79-dd0ba44e34e9.png)

I tried fixing this issue, but can't find a way, would you please have a look @pzelasko , Thanks!

**Comment by pkufool** [comment]
I am a little confused about the AugmentFn in save_audios, how should I use this argument.

**Comment by csukuangfj** [comment]
I think you need to change
https://github.com/lhotse-speech/lhotse/blob/6e06ee85fc8772d51d9300a5e1c577f0553034da/lhotse/cut/base.py#L623-L625

```python3
 torchaudio.save( 
     str(storage_path), torch.as_tensor(samples), sample_rate=self.sampling_rate, encoding="PCM_S", bits_per_sample=16
 ) 
```

**Comment by pzelasko** [comment]
Thanks for reporting this!

The extra arguments should probably be conditional on the extension. E.g. if it's `.wav`, use what @csukuangfj suggested. If it's `.flac`, use sth else; etc. Could you make a PR to address this?

**Comment by pkufool** [comment]
Closed by #896
